l2metrics/report.py

```diff
@@ -561,6 +565,17 @@ def calculate_task_metrics(self) -> None:
                         self.task_metrics_df.at[task, "recovery_times"] = list(
                             tm["recovery_time"].dropna().to_numpy(dtype=float)
                         )
+                    elif metric in ["avg_train_perf", "avg_eval_perf"]:
+                        if metric == "avg_train_perf":
+                            train_perf = self._metrics_df[
```

Comment:
I think these might need to be using the tm dataframe as opposed to self._metrics_df. Otherwise each task is going to get the same mean performance value

Reply:
Just confirmed that each task receives the same values for mean performance by looking at the metrics JSON